### PR TITLE
Add mac customization

### PR DESCRIPTION
**Pull request by philipandag**

### CODE REVIEW

**Comment by philipandag** [comment]
I have missed it but it seems an automatic formatter messed around a bit, for example:
```
  if [[ -n $long_usage ]]; then
    printf "dcu logo - Insert custom logo boot splash into firmware image\n"
    echo

  else
    printf "dcu logo - Insert custom logo boot splash into firmware image\n"
    echo

  fi
```
changed to:
```
  printf "dcu logo - Insert custom logo boot splash into firmware image\n\n"
```

Should I restore those changes?

File: src/mac_command.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  "${IFDTOOL}" -i gbe:flashregion_3_gbe.bin "${DASHARO_ROM}" || { echo "Failed to insert gbe to the binary" ; return 1; }
+  echo "Moving ${DASHARO_ROM}.new to ${DASHARO_ROM}"
+  mv "${DASHARO_ROM}.new" "${DASHARO_ROM}" -f
+  echo "Success"
+}
+
+get_mac() {
+  # dump sections
+  "${IFDTOOL}" -x "${DASHARO_ROM}" > /dev/null 2>&1 || { echo "Failed to extract sections" ; return 1; }
+  "${NVMTOOL}" "flashregion_3_gbe.bin" dump
```
Comment (mkopec) [inline src/mac_command.sh:40]: Not all binaries have the gbe region. Let's add a check if this region exists. Inevitably someone will try to use this tool on an unsupported binary, we should print an error when that happens.

File: src/mac_command.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+DASHARO_ROM="${args[dasharo_rom_file]}"
+MAC="${args[--set]}"
+
+
+
+set_mac() {
+  local _mac="$1"
+
+  "${IFDTOOL}" -x "${DASHARO_ROM}" > /dev/null 2>&1 || { echo "Failed to extract sections" ; return 1; }
+
```
Comment (mkopec) [inline src/mac_command.sh:16]: a/a, let's add a check if the file `flashregion_3_gbe.bin` actually exists

File: src/mac_command.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+GBE_FLASHREGION_FILENAME="flashregion_3_gbe.bin"
+
+
+set_mac() {
+  local _mac="$1"
+
+  "${IFDTOOL}" -x "${DASHARO_ROM}" > /dev/null 2>&1 || { echo "Failed to extract sections" ; return 1; }
+
+  if [[ ! -f "$GBE_FLASHREGION_FILENAME" ]]; then
+    echo "Setting the MAC address inf this binary is currently not supported"
```
Comment (mkopec) [inline src/mac_command.sh:18]:
```suggestion
    echo "Setting the MAC address in this binary is currently not supported"
```

File: src/mac_command.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  "${IFDTOOL}" -i gbe:"$GBE_FLASHREGION_FILENAME" "${DASHARO_ROM}" || { echo "Failed to insert gbe to the binary" ; return 1; }
+  echo "Moving ${DASHARO_ROM}.new to ${DASHARO_ROM}"
+  mv "${DASHARO_ROM}.new" "${DASHARO_ROM}" -f
+  echo "Success"
+}
+
+get_mac() {
+  # dump sections
+  "${IFDTOOL}" -x "${DASHARO_ROM}" > /dev/null 2>&1 || { echo "Failed to extract sections" ; return 1; }
+  "${NVMTOOL}" "$GBE_FLASHREGION_FILENAME" dump
```
Comment (mkopec) [inline src/mac_command.sh:45]: we also need to check if the file exists here

File: src/mac_command.sh (hunk 3 above)
Comment (philipandag) [reply src/mac_command.sh]: commited [34be5bd](https://github.com/Dasharo/dcu/pull/10/commits/34be5bdc4c383d9395d1a9b7522ca1bdc3083908)

File: src/mac_command.sh (hunk 4 above)
Comment (philipandag) [reply src/mac_command.sh:36]: accounted for here:[dd54e9c](https://github.com/Dasharo/dcu/pull/10/commits/dd54e9c0d51d43b1dbbb5831d4cfd11ded9708e9)

**Review by mkopec — APPROVED** [review]
Looks good now, thanks

**Comment by mkopec** [comment]
Merged (forgot to push this branch before fastforwarding `main`, so github didn't count is as a merge...)

**Pull request closed without merging**